server/task_check_scheduler.go

```diff
@@ -179,6 +179,74 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 
 // ScheduleCheckIfNeeded schedules a check if needed.
 func (s *TaskCheckScheduler) ScheduleCheckIfNeeded(ctx context.Context, task *api.Task, creatorID int, skipIfAlreadyTerminated bool) (*api.Task, error) {
+	// All task should pass timing task check.
+	// Since Time is an auto-increment value, a task would generate a timing task check every time before executing.
+	// However, this is an annoying behavior, and we adopt the following logic to ameliorate this problem.
+	{
+		// fetching history task check
+		var err error
```

Comment:
We can have a short circuit logic here to skip the check if task.NotBeforeTs is 0, this is more consistent with we set its default value to 0.